software/metax/misc/Genomics.py

```diff
@@ -5,16 +5,38 @@ def coordinate_format(format, chromosome, position, ref_allele, alt_allele):
     return format.format(chromosome, position, ref_allele, alt_allele)
 
 def map_on_the_fly(mapping, format, chromosome, position, ref_allele, alt_allele):
+    nt = {"A":"T","C":"G", "G":"C","T":"A"}
     r = None
     v = format.format(chromosome, position, ref_allele, alt_allele)
 
     if v in mapping:
-        r = mapping[v]
+        if type(mapping) == dict:
+            r = mapping[v]
+        else:
+            r = v
     elif len(ref_allele) == 1 and len(alt_allele) == 1:
         #try swapped
-        v = format.format(chromosome, position, alt_allele, ref_allele)
-        if v in mapping:
-            r = mapping[v]
+        v_ = format.format(chromosome, position, alt_allele, ref_allele)
+        #try complement
+        c = format.format(chromosome, position, nt[ref_allele], nt[alt_allele])
+        # try swapped complement
+        c_ = format.format(chromosome, position, nt[alt_allele], nt[ref_allele])
+        if v_ in mapping:
```

Comment:
This is confusing - it uses mapping as a mapping or as a filter based on type - which goes against the method name.
I would simply enforce mapping to be a properly built dictionary (asserting it's dict at the method beginning and removing the type check).

Also, the set seems to be initialized to `rsid` -why not use `varID` which actually contains the formatted ids in gtex models?

Reply:
The mapping is used as a filter in this case as it was implemented before when the mapping is a dict object. But since from the prior step I have my mapping as a set that is why I added additional type check. 

The set is initialized to `rsid` because the `rsid` is update to the value passed to `--model_db_snp_key`. If you pass in the `varID` to the argument then that is what will be used as the `rsid`.

Reply:
Ok, thanks for the clarification.
To create a proper dict, I would load the `extra` table again with a dict where, for each row, `varID` column is the key and `rsid` is the value. Then the check for complements becomes more straightforward.